public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -191,7 +185,20 @@ function (angular, _) {
           });
         }
 
-        return this.getDimensionValues(region, namespace, metricName, dimensions);
+        return this.getDimensionValues(region, namespace, metricName, dimensions).then(function(result) {
+          return _.map(result, function(dimensions) {
+            var values = _.chain(dimensions)
+            .sortBy(function(dimension) {
+              return dimension.Name;
+            })
+            .map(function(dimension) {
+              return dimension.Name + '=' + dimension.Value;
```

Comment:
@torkelo 
I agree with your opinion for design perspective.
The reason why I change response format is, I want to support flexible templating.
Without returning dimension name, we can't pass dimesion to GetMetricStatistics() correctly.

The dimension name/value pair is likely primary key of database.
For EC2, the primary key is InstanceId.
For ElastiCache, the primary keys are CacheClusterId and CacheNodeId.
(Please see https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/CacheMetrics.DimensionsAndSets.html)

To get metrics of ElastiCache, we need to specify both key.

I want to generate each cache node graphs in specified cache cluster, by Panel repeat.
If query editor can parse a template value, the format is "dim_key1=dim_value1,dim_key2=dim_value2...", it is possible.